R/SIRD_plot.R

```diff
@@ -1,121 +1,158 @@
+        theme(plot.title = element_text(size = 22, face = "bold")) +
+        theme(legend.position = "bottom") +
+        ylab("Number of People") +
+        xlab("Time") +
+        scale_x_continuous(expand = c(0, 0)) +
+        scale_y_continuous(expand = c(0, 0)) +
+        geom_line(aes(y = S, color = "Blue"), linewidth = 1.5) +
+        geom_line(aes(y = I, color = "Red"), linewidth = 1.5) +
+        geom_line(aes(y = R, color = "Green"), linewidth = 1.5) +
+        geom_line(aes(y = D, color = "Orange"), linewidth = 1.5) +
+        scale_color_identity(
             name = "SIRD", breaks = c("Blue", "Red", "Green", "Orange"),
             labels = c("Susceptible", "Infected", "Recovered", "Dead"), guide = "legend"
-          )
-      })
-    })
+        )
+}
+
 
-    output$SIRDPhasePlane <- renderPlot({
-      input$go
-      isolate({
-        val <- as.data.frame(sird_values())
-        ggplot(val, aes(x = S)) +
-          geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
-          theme(
-            axis.line = element_line(color = "black"), axis.text = element_text(size = 14),
+plotPhasePlaneSIRD <- function() {
+    valuesSIRD <- solveSIRD()
+    ggplot(valuesSIRD, aes(x = S)) +
+        geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
+        theme(
+            axis.line = element_line(color = "black"),
+            axis.text = element_text(size = 14),
             axis.title.x = element_text(size = 16, face = "bold"),
             axis.title.y = element_text(size = 16, face = "bold")
-          ) +
-          theme(legend.position = "bottom") +
-          ggtitle("SI Phase Plane") +
-          theme(plot.title = element_text(size = 22, face = "bold")) +
-          ylab("Infected (I)") +
-          scale_x_continuous(expand = c(0, 0)) +
-          scale_y_continuous(expand = c(0, 0)) +
-          xlab("Susceptible (S)") +
-          scale_color_identity(
-            breaks = "Blue",
-            labels = "Susceptible"
-          )
-      })
-    })
+        ) +
+        theme(legend.position = "bottom") +
+        ggtitle("SI Phase Plane") +
+        theme(plot.title = element_text(size = 22, face = "bold")) +
+        ylab("Infected (I)") +
+        scale_x_continuous(expand = c(0, 0)) +
+        scale_y_continuous(expand = c(0, 0)) +
+        xlab("Susceptible (S)") +
+        scale_color_identity(
+            name = "SIRD", breaks = c("Blue", "Red", "Green", "Orange"),
+            labels = c("Susceptible", "Infected", "Recovered", "Dead"), guide = "legend"
+        )
+}
 
-    output$tableSIRD <- renderTable({
-      input$go
-      isolate({
-        val <- as.data.frame(sird_values())
-        val <- val[-c(7)]
-        return(val)
-      })
+# no longer need to do shiny::isRunning() check since this function is only called
+# from server.R
+solveAndRenderSIRD <- function() {
+    expr <- quote({
+        output$plotSIRD <- renderPlot({
+            plotSIRD()
+        })
```

Comment:
Please change this to the following,
```r
output$modelPlot <- renderPlot(plotSIRD())
```
because we no longer have individual outputs for every model type. There is only ever one plot shown for a given model, so it is named `modelPlot`.

Reply:
The braces and line feeds are unnecessary because the `renderPlot` function ultimately consumes a simple value, not a compound expression. You're calling a plotting function, so it's a simple expression.

However, if you wrote an in-line, compound expression to create the object to be rendered as a plot in the Shiny app the braces for a compound expression would become necessary.

```r
shiny::renderPlot({
  ggplot(data = sample_df) +
    geom_point(mapping = aes(group, value)) +
    geom_point(aes(group, group_mean),
              group_means_df,
              colour = 'red',
               size = 3)
})
```